### DISCUSSION TITLE
Unable to Transform Location Header in Proxied Response

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/yarp#1909] [Q&A] [answered]
Hi

I am attempting to transform the location response header after following the "Direct Forwarding" example, with the following code:

![image](https://user-images.githubusercontent.com/56893397/197864932-288f40a3-a460-4c06-a469-da4be48f7db3.png)

I can see from debugging that I am setting the new location here as expected. However, a fiddler trace of the e2e proxied request still has the original location on the response header.

Can someone provide guidance of where I am going wrong?

**Comment by Tratcher** [comment, marked as answer]
The headers have already been copied to HttpContext.Response.Headers, you need to modify the Location there.

**Reply by MihaZupan** [reply]
That, or move the `base.TransformResponseAsync` call after your modification of the `HttpResponseMessage` headers.

**Reply by ghost** [reply]
Thank you both.
